pkg/gui/filetree/collapsed_paths.go

```diff
@@ -1,20 +1,38 @@
 package filetree
 
-type CollapsedPaths map[string]bool
+import "github.com/jesseduffield/generics/set"
 
-func (cp CollapsedPaths) ExpandToPath(path string) {
+type CollapsedPaths struct {
+	collapsedPaths *set.Set[string]
```

Comment:
not proud of calling this field the same as the struct name but calling it `set` sounded similarly wrong. I could have just used the set directly instead of wrapping it in the struct but I like the resulting encapsulation